src/Lecture2.hs

```diff
@@ -177,8 +281,14 @@ False
 >>> isIncreasing [1 .. 10]
 True
 -}
+
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
+isIncreasing [] = True
+isIncreasing [_] = True
+isIncreasing [x,y] = y > x
+isIncreasing (x:rest@(y:_))
+    | y < x = False
+    | otherwise = isIncreasing rest
```

Comment:
You can rely on lazy evaluation in Haskell here because the `&&` function is lazy on the second argument if the left argument returns `False`
```suggestion
isIncreasing (x:rest@(y:_)) = y < x && isIncreasing rest
```